### DISCUSSION TITLE
How can i get if field is focused?

### DISCUSSION THREAD

**Discussion by b00sta** [react-hook-form/react-hook-form#3977] [Q&A]
I couldn't find out how to get if field is focused.

**Comment by bluebill1049** [comment]
share a simple codesandbox.

**Reply by b00sta** [reply]
https://codesandbox.io/s/cool-morning-hi8zo

I want to add custom handling of focused field. how can i get it?

**Reply by bluebill1049** [reply]
https://react-hook-form.com/api#Controller

take a look at the onFocus prop

**Reply by b00sta** [reply]
So i need to manually store/change it in component state? i mean there is no meta prop for that out-of-box.

**Reply by bluebill1049** [reply]
i thought you want to focus? you can set up onFucs with ref and when there is an error, the input will get focused (if ref has onFocus method)

**Reply by b00sta** [reply]
No, i need to for example change style of focused field

**Reply by bluebill1049** [reply]
that's at your component level, not controller level.

**Reply by b00sta** [reply]
Can you provide an example? You can just clone my codesandbox. i left a comment in place where i need to get that state
